OriginateTheme/Scripts/ot_reference_resolver.py

```diff
@@ -0,0 +1,70 @@
+import json
+import os
+import re
+import string
+import sys
+
+def main(argv):
+    """
+        Method responsible for the program execution.
+
+        Parameters
+        ----------
+        argv: Array
+            Array containing all programs' arguments
+    """
+
+    # Extract the inputFile and outputDirectory.
+    (inputFile) = parseArguments(argv)
+
+    # Open the .JSON file and parse the fonts, colors and components.
+    with open(inputFile, 'r') as file:
+        try:
+            data = json.load(file, 'utf-8')
+            resolveComponentReferencesInData(data)
+            inputFile.write(data)
```

Comment:
You are opening the input file with reading permissions but than write on the stream. Does that work here? In addition, I think instead of modifying the original file we should generate a temporary file which will be used for all further processing. Otherwise we gonna modify the templates (which could be extended in the future using variables.)